core/src/test/scala/traveler/CCArenaSpec.scala

```diff
@@ -0,0 +1,31 @@
+package traveler
+
+class CCArenaSpec extends munit.FunSuite:
+  test("shared arena catches leaks"):
+    assertNoDiff(
+      compileErrors {
+        """CCArena.shared{(a: CCSharedArena^) ?=>
+          a.get.toString()
+    }
+    """
+      },
+      ""
+    )
+
+  test("multiple shared arenas are allowed"):
+    CCArena.shared: 
+      val a = CCArena()
+      val r = CCArena.shared:
+        a.get
+      r.toString()
+
+  test("shouldn't compile"):
+    CCArena.confined: 
+      val a = CCArena()
+      val b: CCSharedArena^ = ???
+      val r = NThread.inner(using b):
+        a.get.toString
```

Comment:
a should create an error from being captured when the allowed captureset doesn't include it right?